### PR TITLE
econml 0.15.0 ❄️

### PR DESCRIPTION
**Pull request by psteyer**
econml 0.15.0 ❄️

**Destination channel:** Snowflake

### Links

- [PKG-4217]
- [Upstream repository](https://github.com/py-why/EconML/tree/v0.15.0)
- [Upstream changelog/diff](https://github.com/py-why/EconML/compare/v0.13.1...v0.15.0)

### Changes

- recipe
- upstream tests (a couple of patches to make them run): they are extremely slow

[PKG-4217]: https://anaconda.atlassian.net/browse/PKG-4217?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by danpetry — APPROVED** [review]

**Comment by danpetry** [comment]
Adding upstream tests would be nice if poss: https://github.com/py-why/EconML/tree/v0.15.0/econml/tests

File: recipe/meta.yaml [hunk 1]
```diff
@@ -8,37 +8,33 @@ package:
...
 requirements:
   build:
     - {{ compiler('c') }}
-    - {{ compiler('cxx') }}
-    - python                                 # [build_platform != target_platform]
-    - cross-python_{{ target_platform }}     # [build_platform != target_platform]
-    - cython                                 # [build_platform != target_platform]
-    - numpy                                  # [build_platform != target_platform]
   host:
     - cython
```
Comment (lorepirri) [inline recipe/meta.yaml]: what about
```
    "scipy",
    "cython<3" # Our native code has several incompatibilities with Cython 3
```
https://github.com/py-why/EconML/blob/479362aceb3205b6515adf5084f59a24ea79785d/pyproject.toml#L87-L88

**Review by lorepirri — APPROVED** [review]

**Comment by lorepirri** [comment]
@boldorider4 I'll take on this one as well and add upstream tests

**Review by lorepirri — COMMENTED** [review]
adding upstream tests

**Review by boldorider4 — APPROVED** [review]

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,60 +1,91 @@
...
 
 build:
   number: 0
-  script: {{ PYTHON }} -m pip install . -vv
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+  skip: True  # [s390x or py>311]
+  # Segfault detected on some platforms during testing. This is enough of a reason to skip this arch for now
+  # Unfortunately this segfault is not documented anywhere
+  skip: True  # [linux and aarch64]
+  skip: True  # [osx and 64]
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml:23]:
```suggestion
  skip: True  # [osx]
```
It crashes (real hard kernel crash) our osx-arm64 machines.

Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml]:
```suggestion
  skip: True  # [osx]
```
It crashes (real hard kernel crash) our osx-arm64 machines.

**Review by JeanChristopheMorinPerso — CHANGES_REQUESTED** [review]

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,60 +1,91 @@
...
 
 about:
   home: https://github.com/Microsoft/EconML
   summary: This package contains several methods for calculating Conditional Average Treatment Effects
+  description: |
+    EconML is a Python package for estimating heterogeneous treatment effects from observational data via
+    machine learning. This package was designed and built as part of the ALICE project at Microsoft
+    Research with the goal to combine state-of-the-art machine learning techniques with econometrics to 
+    bring automation to complex causal inference problems.
   license: BSD-3-Clause
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]:
```suggestion
  license: MIT AND BSD-3-Clause
```
see https://github.com/py-why/EconML/blob/v0.15.0/LICENSE

File: recipe/meta.yaml [hunk 4]
```diff
@@ -1,60 +1,91 @@
...
+  license_family: BSD
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]:
```suggestion
  license_family: Other
```

File: recipe/meta.yaml [hunk 5]
```diff
@@ -1,60 +1,91 @@
...
-    - {{ compiler('cxx') }}
-    - python                                 # [build_platform != target_platform]
-    - cross-python_{{ target_platform }}     # [build_platform != target_platform]
-    - cython                                 # [build_platform != target_platform]
-    - numpy                                  # [build_platform != target_platform]
+    - patch       # [not win]
+    - m2-patch    # [win]
   host:
-    - cython
+    - cython 0.29.36
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]: Wouldn't it be better to use `cython 0.29`?

File: recipe/meta.yaml [hunk 6]
```diff
@@ -1,60 +1,91 @@
...
+    - setuptools
   run:
-    - dowhy <0.8
     - lightgbm
-    - python-graphviz
     - joblib >=0.13.0
-    - numba !=0.42.1
     - numpy
-    - pandas
+    - numba
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:44]: Could you comment that `numba` is required here https://github.com/py-why/EconML/blob/v0.15.0/econml/solutions/causal_analysis/_causal_analysis.py#L11?

File: recipe/meta.yaml (hunk 5 above)
Comment (boldorider4) [reply recipe/meta.yaml]: I don't see why it would be better. `0.29.36` is available for all platforms and all pyver.

Comment (JeanChristopheMorinPerso) [reply recipe/meta.yaml]: @boldorider4 It's for the future. Pinning to "0.29.36" doesn't bring any benefit and "0.29" is enough.

**Review by JeanChristopheMorinPerso — APPROVED** [review]

**Review by skupr-anaconda — APPROVED** [review]

**Pull request merged**